### PR TITLE
feat(hardware): add InstrumentInfoRequest to get info from Pipettes & Gripper

### PR DESCRIPTION
**Pull request by ahiuchingau**
# Overview
Since pipette and gripper are technically both robot instrument, I'm adding an InstrumentInfoRequest so both systems can respond to host when prompted. 

This also means we have to update the tool detector class to read the info from both instruments correctly.

# Changelog

# Review requests

# Risk assessment

### CODE REVIEW

File: hardware/opentrons_hardware/hardware_control/tools/detector.py [hunk 1]
```diff
@@ -78,19 +82,22 @@ def __init__(self, messenger: CanMessenger) -> None:
     @staticmethod
     async def _await_responses(
         callback: WaitableCallback, for_nodes: Set[NodeId]
-    ) -> Dict[NodeId, PipetteInformation]:
-        to_ret: Dict[NodeId, PipetteInformation] = {}
+    ) -> Tuple[Dict[NodeId, PipetteInformation], Dict[NodeId, GripperInformation]]:
```
Comment (sfoster1) [inline hardware/opentrons_hardware/hardware_control/tools/detector.py:85]: can we make a `TypedDict` spec here, optional keys, pipette (and bootloader) node is to pipette info and gripper (and bootloader) node to gripper info?

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Looks great but can we swap the types a bit

File: hardware/opentrons_hardware/hardware_control/tools/detector.py (hunk 1 above)
Comment (sfoster1) [reply hardware/opentrons_hardware/hardware_control/tools/detector.py:85]: Nevermind - mypy can't do this kind of value constraint

**Review by sfoster1 — APPROVED** [review]
Awesome! Let's just drop in an integration test.

**Comment by ahiuchingau** [comment]
Opened a new [ticket](https://opentrons.atlassian.net/browse/RET-1040?atlOrigin=eyJpIjoiZTExYzk5NzJjNTcwNGViN2I4N2I3MGJkYzhlNDFjODAiLCJwIjoiaiJ9) on Jira for adding an integration test

**Pull request merged**